client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -574,8 +574,13 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
             partitionType match {
               case PartitionType.MAP =>
                 if (response.getStatus == StatusCode.SUCCESS.getValue) {
+                  var partitionLocationsList = response.getPartitionLocationsList
+                  if (CollectionUtils.isEmpty(partitionLocationsList)) {
+                    partitionLocationsList =
+                      response.getPartitionLocationList.getPartitionLocationsList
+                  }
                   val partitionLocations =
-                    response.getPartitionLocationsList.asScala.filter(
```

Comment:
```
  private java.util.List<org.apache.celeborn.common.protocol.PbPartitionLocation> partitionLocations_;
  /**
   * <code>repeated .PbPartitionLocation partitionLocations = 2;</code>
   */
  @java.lang.Override
  public java.util.List<org.apache.celeborn.common.protocol.PbPartitionLocation> getPartitionLocationsList() {
    return partitionLocations_;
  }
```

Checkout the PB-generated code snippet. I don't think here is a performance issue.